app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -80,6 +80,7 @@ def analytics_properties
         context: context,
         multi_factor_auth_method: params[:otp_delivery_preference],
         confirmation_for_add_phone: confirmation_for_add_phone?,
+        mfa_id: user_session[:phone_id],
```

Comment:
I think it's potentially confusing to have `mfa_id` potentially referring to different records and have to disambiguate via `multi_factor_auth_method`.

What do you think of making the keys more descriptive in a way that ensures they are unique and refer only to a single record? e.g. `phone_configuration_id`, `piv_cac_id`, `webauthn_configuration_id`, `auth_app_configuration_id`

Reply:
Not sure why it is confusing.  It is the exact way rails does polymorphic types https://guides.rubyonrails.org/association_basics.html#polymorphic-associations.   There is a column for id and a free text column for type. Also, it's all structured to query for say just piv_cac_ids.

Reply:
if you both still feel strongly about it I'll change it

Reply:
The distinction for me is Rails has an explicit construct with `polymorphic: true`, and code to automatically build the queries so end users don't have to figure out `piv_cac_ids`. CloudWatch and the Analytics class don't have a way to hint that these two fields are related and do the conversion for you.

I do feel somewhat strongly that my initial suggestion helps the logs be more intuitive 🙂

Reply:
ok @zachmargolis will be pleased :-)

Reply:
Yes I will be pleased, thank you 😁